tpu-client-next/src/connection_workers_scheduler.rs

```diff
@@ -21,14 +21,19 @@ use {
     tokio::sync::mpsc,
     tokio_util::sync::CancellationToken,
 };
+pub type TransactionReceiver = mpsc::Receiver<TransactionBatch>;
 
 /// The [`ConnectionWorkersScheduler`] sends transactions from the provided
 /// receiver channel to upcoming leaders. It obtains information about future
 /// leaders from the implementation of the [`LeaderUpdater`] trait.
 ///
 /// Internally, it enables the management and coordination of multiple network
 /// connections, schedules and oversees connection workers.
-pub struct ConnectionWorkersScheduler;
+pub struct ConnectionWorkersScheduler {
+    leader_updater: Box<dyn LeaderUpdater>,
+    transaction_receiver: TransactionReceiver,
+    pub stats: Arc<SendTransactionStats>,
```

Comment:
I would prefer having a method for that instead of making it pub. Because as soon as we publish this crate, it will be very hard to change the implementation because this `pub stats` will be part of public API.